tests/test_service.py

```diff
@@ -0,0 +1,368 @@
+from PyCrypCli.client import Client
+from PyCrypCli.exceptions import (
+    AlreadyOwnThisServiceException,
+    CannotDeleteEnforcedServiceException,
+    CannotToggleDirectlyException,
+    CouldNotStartService,
+    DeviceNotFoundException,
+    DeviceNotOnlineException,
+    InvalidRequestException,
+    PermissionDeniedException,
+    ServiceCannotBeUsedException,
+    ServiceNotFoundException,
+    ServiceNotSupportedException,
+    WalletNotFoundException,
+)
+
+from database import execute
+from testcase import TestCase
+from tests.test_device import setup_device
+from tests.test_server import setup_account, super_password, super_uuid
+from util import get_client, uuid
+
+
+def create_service(device=uuid(), name="telnet", n=1, part_owner=None, owner=super_uuid, clear_service=True):
+    if clear_service:
+        clear_services()
+
+    service_uuids = []
+    for i in range(n):
+        service_uuids.append(uuid())
+        execute(
+            "INSERT INTO service_service(uuid, device, owner, name, running,running_port, part_owner)"
+            + "VALUES (%s,%s,%s,%s,%s,%s,%s)",
+            service_uuids[i],
+            device,
+            owner,
+            name,
+            i % 2 == 0,
+            1337,
+            part_owner,
+        )
+
+    return service_uuids
+
+
+def clear_services():
+    execute("TRUNCATE service_service")
+
+
+class TestService(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        setup_account()
+        cls.client: Client = get_client()
+        cls.client.login("super", super_password)
+
+    @classmethod
+    def tearDownClass(cls: "TestService"):
+        cls.client.close()
+
+    def test_public_info_successful(self):
+        device_uuid = setup_device()[0]
+        service_uuid = create_service(device_uuid)[0]
+        actual = self.client.ms("service", ["public_info"], device_uuid=device_uuid, service_uuid=service_uuid)
```

Comment:
Please leave an empty line between setup and assertions to improve readability. (applies to other test cases as well)